internal/server/evaluation/legacy_evaluator_test.go

```diff
@@ -354,6 +401,64 @@ func Test_matchesNumber(t *testing.T) {
 				Value:    "bar",
 			},
 		},
+		{
+			name: "is one of",
+			constraint: storage.EvaluationConstraint{
+				Property: "foo",
+				Operator: "isoneof",
+				Value:    "[3, 3.14159, 4]",
+			},
+			value:     "3.14159",
+			wantMatch: true,
+		},
+		{
+			name: "negative is one of",
+			constraint: storage.EvaluationConstraint{
+				Property: "foo",
+				Operator: "isoneof",
+				Value:    "[5, 3.14159, 4]",
+			},
+			value: "9",
+		},
+		{
+			name: "negative is one of (invalid json)",
+			constraint: storage.EvaluationConstraint{
+				Property: "foo",
+				Operator: "isoneof",
+				Value:    "[5, \"str\"]",
```

Comment:
Hmm this is valid JSON though correct?

Reply:
its valid JSON just invalid for Flipt because it's mixed number and string

Reply:
I've updated the description to say "non-number values" instead.